src/tree/BranchDataProviderItem.ts

```diff
@@ -110,5 +115,11 @@ export class BranchDataItemWrapper implements ResourceGroupsItem, Wrapper {
 export type BranchDataItemFactory = (branchItem: ResourceModelBase, branchDataProvider: BranchDataProvider<ResourceBase, ResourceModelBase>, options?: BranchDataItemOptions) => BranchDataItemWrapper;
 
 export function createBranchDataItemFactory(itemCache: BranchDataItemCache): BranchDataItemFactory {
-    return (branchItem, branchDataProvider, options) => new BranchDataItemWrapper(branchItem, branchDataProvider, itemCache, options);
+    return (branchItem, branchDataProvider, options) => {
+        const cachedItem = itemCache.getItemForId(branchItem.id);
```

Comment:
Upon refresh what clears the caches?

Reply:
When the extension host called `getChildren(undefined)` the cache is cleared.

https://github.com/microsoft/vscode-azureresourcegroups/blob/628de8437a48bfaff8f46e66dac647fb26c971f1/src/tree/wrapTreeForVSCode.ts#L15

Reply:
So only a full refresh can clear the cache--no more partials?

Reply:
Yes. And just to clarify, I haven't changed that behavior in this PR.